airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityImpl.java

```diff
@@ -128,25 +179,27 @@ public GeneratedJobInput getSyncWorkflowInput(final SyncInput input) {
           .withNormalizationDockerImage(destinationNormalizationDockerImage)
           .withSupportsDbt(destinationDefinition.getSupportsDbt())
           .withNormalizationIntegrationType(normalizationIntegrationType)
-          .withAllowedHosts(configReplacer.getAllowedHosts(destinationDefinition.getAllowedHosts(), config.getDestinationConfiguration()));
+          .withAllowedHosts(configReplacer.getAllowedHosts(destinationDefinition.getAllowedHosts(), attemptSyncConfig.getDestinationConfiguration()));
 
       final StandardSyncInput syncInput = new StandardSyncInput()
           .withNamespaceDefinition(config.getNamespaceDefinition())
           .withNamespaceFormat(config.getNamespaceFormat())
           .withPrefix(config.getPrefix())
           .withSourceId(standardSync.getSourceId())
           .withDestinationId(standardSync.getDestinationId())
-          .withSourceConfiguration(config.getSourceConfiguration())
-          .withDestinationConfiguration(config.getDestinationConfiguration())
+          .withSourceConfiguration(attemptSyncConfig.getSourceConfiguration())
+          .withDestinationConfiguration(attemptSyncConfig.getDestinationConfiguration())
```

Comment:
all references that previously used the JobSyncConfig to retrieve config and state now move to using the attemptSyncConfig